documentation/gnmi/design/sonic-dash.yang

```diff
@@ -0,0 +1,490 @@
+                    type string {
+                        length 1..255;
+                    }
+                }
+
+                leaf-list address_spaces {
+                     type union {
+                         type stypes:sonic-ip4-prefix;
+                         type stypes:sonic-ip6-prefix;
+                     }
+                }
+
+            } /* end of list DASH_VNET_LIST */
+        } /* end of container DASH_VNET */
+
+        container DASH_QOS {
+            description "DASH QOS configs to specify bandwidth, cps, part of APP_DB";            
+
+            list DASH_QOS_LIST {
+                key "name";
+
+                leaf name{
+                    type string {
+                        length 1..255;
+                    }
+                }
+
+                leaf bw {
+                    description "Bandwidth in kbps"
+                    type uint64 {
+                        range "0..100000000" {
+                            error-message "kbps value out of range";
+                        }
+                    }
+                }
+
+                leaf cps {
+                    description "Max number of connections per second"
+                    type uint64 {
+                        range "0..100000000" {
+                            error-message "cps value out of range";
+                        }
+                    }
+                }
+
+                leaf flows {
+                    description "Number of flows"
+                    type uint64 {
+                        range "0..100000000" {
+                            error-message "flows value out of range";
+                        }
+                    }
+                }
+
+            } /* end of list DASH_QOS_LIST */
+        } /* end of container DASH_QOS */
+        
+        container DASH_ENI {
+            description "DASH ENI object to specify eni params; part of APP_DB";            
+
+            list DASH_ENI_LIST {
+                key "name";
+
+                leaf name{
+                    type string {
+                        length 1..255;
+                    }
+                }
+
+                leaf eni_id{
+                    type string {
+                        length 1..255;
+                    }
+                }
+
+                leaf mac_address {
+                    type yang:mac-address;
+                }
+
+                leaf qos {
+                    type leafref {
+                        path /dash:sonic-dash/dash:DASH_QOS/dash:DASH_QOS_LIST/dash:name;
+                    }
+                }
+
+                leaf-list vnet {
+                    type leafref {
+                        path /dash:sonic-dash/dash:DASH_VNET/dash:DASH_VNET_LIST/dash:name;
+                    }
+                }
+
+            } /* end of list DASH_ENI_LIST */
+        } /* end of container DASH_ENI */
+     
+
+        container DASH_ACL_V4_IN {
```

Comment:
DASH_ACL_V4_IN -> DASH_ACL_IN?
DASH_ACL_V4_OUT -> DASH_ACL_OUT
Can't we use single table for both V4 & V6 ACL types since we have ip_version in ACL_GROUP table?